src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +62,801 @@ const ActorInit En_Famos_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AD888.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AD8B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADA74.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADAE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADB4C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADB70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADC40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADC64.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADD20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADDA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADE00.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADE74.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADFA4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADFF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE030.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/EnFamos_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE304.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE3A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE3FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/EnFamos_Draw.s")
+static s32 animatedMaterialsDesgmented = false;
+
+void EnFamos_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFamos* this = THIS;
+    Path* path;
+    // prob fake match
+    s32 sTrue = 1; // required to be separate to match
+    int i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if (GET_FAMOS_PATH(thisx) != 0xFF) {
+        path = &globalCtx->setupPathList[this->actor.params];
+        this->pathPoints = Lib_SegmentedToVirtual(path->points);
+        this->pathNodeCount = path->count;
+        if (this->pathNodeCount == 1) {
+            this->pathPoints = NULL;
+            this->pathNodeCount = 0;
+        }
+    }
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawSquare, 30.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gFamosSkeleton, &gFamosLowerSNSAnim, this->jointTable,
+                   this->morphTable, FAMOS_LIMB_MAX);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider1, &this->actor, &sCylinderInit1);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider2, &this->actor, &sCylinderInit2);
+    Collider_InitAndSetJntSph(globalCtx, &this->emblemCollider, &this->actor, &sJntSphInit,
+                              &this->emblemColliderElements);
+
+    if (!animatedMaterialsDesgmented) { // init animated materials
```

Comment:
```suggestion
    if (!sAnimatedMaterialsDesegmented) { // init animated materials
```
Should have an `s` prefix still for `static`, as well as small typo in `desegmented`